src/ex_cmds2.c

```diff
@@ -181,10 +181,30 @@ dialog_changed(
 	// May get file name, when there is none
 	browse_save_fname(buf);
 #endif
-	if (buf->b_fname != NULL && check_overwrite(&ea, buf,
-				    buf->b_fname, buf->b_ffname, FALSE) == OK)
+	int empty_buf = buf->b_fname == NULL ? TRUE : FALSE;
+	if (empty_buf == TRUE)
+	{
+	    buf->b_fname = vim_strsave((char_u *)"Untitled");
+	    if (STRLEN(NameBuff) == 0 && mch_dirname(NameBuff, MAXPATHL) == OK)
```

Comment:
I don't understand this. What is the reason for the `STRLEN()`? and why do you copy only the current directory into this buffer?

Reply:
sometimes  NameBuff is empty for me after run :pwd it has value ..

Reply:
without `STRLEN` check the `Test_goto_buf_with_confirm` will failed.